deepcell_toolbox/deep_watershed.py

```diff
@@ -101,55 +101,81 @@ def deep_watershed(outputs,
     based watershed of the pixelwise interior prediction.
 
     Args:
-        outputs (list): DeepWatershed model output. A list of
-            [inner_distance, outer_distance, fgbg, pixelwise].
+        model_output (dict): DeepWatershed model output. A dictionary containing key: value pairs
+            with the transform name and the corresponding output. Currently supported keys:
 
             - inner_distance: Prediction for the inner distance transform.
             - outer_distance: Prediction for the outer distance transform.
-            - fgbg: Prediction for the foregound/background transform.
-            - pixelwise: Prediction for the interior/border/background transform.
+            - fgbg: Foreground prediction for the foregound/background transform.
+            - pixelwise_interior: Interior prediction for the interior/border/background transform.
 
-        min_distance (int): Minimum allowable distance between two cells.
-        detection_threshold (float): Threshold for the inner distance.
-        distance_threshold (float): Threshold for the outer distance.
-        exclude_border (bool): Whether to include centroid detections
-            at the border.
+        min_distance (int): Minimum allowable distance between two cell maxima.
+        maxima_threshold (float): Threshold for the maxima prediction.
+        cell_threshold (float): Threshold for the cell prediction.
+        exclude_border (bool): Whether to include centroid detections at the border.
         small_objects_threshold (int): Removes objects smaller than this size.
+        cell_model: semantic head to use to predict location of cells
+        maxima_model: semantic head to use to predict maxima of cells
+        cell_model_smooth: smoothing factor to apply to interior model predictions
+        maxima_model_smooth: smoothing factor to apply to interior model predictions
 
     Returns:
         numpy.array: Uniquely labeled mask.
     """
-    inner_distance_batch = outputs[0][:, ..., 0]
-    pixel_interior_batch = outputs[3][:, ..., 1]
+
+    valid_model_names = {'inner-distance', 'outer-distance', 'fgbg-fg', 'pixelwise-interior'}
+
+    if cell_model not in valid_model_names:
+        raise ValueError('interior_model must be one of {}, got {}'.format(valid_model_names,
+                                                                           cell_model))
+
+    if maxima_model not in valid_model_names:
+        raise ValueError('maxima_model must be one of {}, got {}'.format(valid_model_names,
+                                                                         maxima_model))
+
+    cell_prediction_batch = model_output[cell_model]
+    maxima_prediction_batch = model_output[maxima_model]
+
+    if len(cell_prediction_batch.shape) != 4:
+        raise ValueError('Model output must be of length 4. The cell_prediction model '
+                         'provided was of shape {}'.format(cell_prediction_batch.shape))
```

Comment:
Maybe a loop since the text is so similar?

```
zipped = zip(['cell_prediction', 'maxima_prediction'], (cell_prediction_batch, maxima_prediction_batch))
for name, arr in zipped:
    if len(arr.shape) !=4:
        raise ValueError(...)
```